src/NHibernate/Type/DbTimestampType.cs

```diff
@@ -46,47 +45,55 @@ public override object Seed(ISessionImplementor session)
 			}
 		}
 
-		private object GetCurrentTimestamp(ISessionImplementor session)
+		protected virtual DateTime GetCurrentTimestamp(ISessionImplementor session)
 		{
-			Dialect.Dialect dialect = session.Factory.Dialect;
-			string timestampSelectString = dialect.CurrentTimestampSelectString;
-			return UsePreparedStatement(timestampSelectString, session);
+			var dialect = session.Factory.Dialect;
+			// Need to round notably for Sql Server DateTime with Odbc, which has a 3.33ms resolution,
+			// causing stale data update failure 2/3 of times if not rounded to 10ms.
+			return Round(
+				UsePreparedStatement(dialect.CurrentTimestampSelectString, session),
```

Comment:
In fact I am fixing there the trouble which was causing Odbc to be ignored for NH-1756 tests and DbVersionFixture. Going to re-enable them.